### PR TITLE
BUTTON MERGE

### PR DESCRIPTION
**Pull request by Romguil**
feature "button merge"

### CODE REVIEW

File: public/app/HistoryCtrl.js [hunk 1]
```diff
@@ -7,5 +7,15 @@ angular.module('FileSync').controller('HistoryCtrl', ['HistoryService', 'Visibil
...
       HistoryService.remove(edit);
     };
+
+    this.merge = function (edit) {
+      console.log(this.mergePath);
+      if (this.mergePath===undefined){
+        alert("vous devez saisir le chemin pour merger");
+      }
+      else{
+      HistoryService.merge(edit,this.mergePath);
```
Comment (FGRibreau) [inline public/app/HistoryCtrl.js:17]: c'est le rôle de HistoryService.merge() de vérifier que mergePath est valide ou non

File: public/app/HistoryService.js [hunk 2]
```diff
@@ -3,11 +3,12 @@ angular.module('FileSync')
...
     var edits = [];
 
-    SocketIOService.onFileChanged(function (filename, timestamp, content) {
+    SocketIOService.onFileChanged(function (filename, timestamp, content, filepath) {
       edits.unshift({
         filename: filename,
         timestamp: timestamp,
-        content: content
+        content: content,
+		    filepath: filepath
```
Comment (FGRibreau) [inline public/app/HistoryService.js:11]: identation

File: server.js [hunk 3]
```diff
@@ -46,6 +47,12 @@ sio.on('connection', function (socket) {
     socket.visibility = state;
     sio.emit('users:visibility-states', getVisibilityCounts());
   });
+  
+  socket.on('user-merge', function (edit,filepath) {
+	  fs.writeFileSync(filepath, edit.content,'utf8');
```
Comment (FGRibreau) [inline server.js:52]: ahhh mais il faut utiliser la HTML5 Writer API.
Là ça fonctionnait parce que le server.js était sur ta propre machine, mais ça ne sera pas le cas en production ! :)

Comment (Romguil) [inline server.js:52]: je l'avais fait, j'aurais dû le laisser en commentaire

/_function onInitFsWrite(fs) {
            //fs.root.getFile('‪C:\Users\Romain\Desktop\filesync-master\test\test3.txt', {create: true}, writeFile, errorHandler);
            fs.root.getFile(edit.filepath, {create: true}, writeFile, errorHandler);
            console.log('Systeme de fichier: ' + fs.name);
        }
        function onInitFsRead(fs) {
            //fs.root.getFile('‪C:\Users\Romain\Desktop\filesync-master\test\test3.txt', {create: true}, readFile, errorHandler);
            fs.root.getFile(edit.filepath, {create: true}, readFile, errorHandler);
            console.log('Systeme de fichier: ' + fs.name);
        }
        function onInitFsDelete(fs) {
            //fs.root.getFile('‪C:\Users\Romain\Desktop\filesync-master\test\test3.txt', {create: false}, deleteFile, errorHandler);
            fs.root.getFile(edit.filepath, {create: false}, deleteFile, errorHandler);
            console.log('Systeme de fichier: ' + fs.name);
        }
        function readFile(fileEntry) {
            fileEntry.file(function(file) {
                var reader = new FileReader();
                reader.onloadend = function(e) {
                    alert(this.result);
                };
                reader.readAsText(file);
            }, errorHandler);
        }
        function deleteFile(fileEntry) {
            fileEntry.remove(function() {
                alert('delete OK');
            }, errorHandler);
        }  
        function writeFile(fileEntry) {
            fileEntry.createWriter(function(fileWriter) {
                fileWriter.onwriteend = function(e) {
                    alert('ecriture OK.');
                };
                fileWriter.onerror = function(e) {
                    alert('Error: ' + e.toString());
                };
                var data = new Blob([edit.content], {type: 'text/plain'});
                fileWriter.write(data);
            }, errorHandler);
        }
        function errorHandler(e) {
            var msg = '';
            switch (e.code) {
                case FileError.QUOTA_EXCEEDED_ERR:
                    msg = 'QUOTA_EXCEEDED_ERR';
                    break;
                case FileError.NOT_FOUND_ERR:
                    msg = 'NOT_FOUND_ERR';
                    break;
                case FileError.SECURITY_ERR:
                    msg = 'SECURITY_ERR';
                    break;
                case FileError.INVALID_MODIFICATION_ERR:
                    msg = 'INVALID_MODIFICATION_ERR';
                    break;
                case FileError.INVALID_STATE_ERR:
                    msg = 'INVALID_STATE_ERR';
                    break;
                default:
                    msg = 'Unknown Error';
                    break;
            };
            alert('Error: ' + msg);
        }
        window.requestFileSystem  = window.requestFileSystem || window.webkitRequestFileSystem;
        //window.requestFileSystem(window.TEMPORARY, 1024_1024, onInitFsDelete, errorHandler);
        window.requestFileSystem(window.TEMPORARY, 1024_1024, onInitFsWrite, errorHandler);
        window.requestFileSystem(window.TEMPORARY, 1024_1024, onInitFsRead, errorHandler); */

**Pull request closed without merging**